public/pages/workflow_detail/agentic_search/test_flow/test_flow.tsx

```diff
@@ -122,6 +125,13 @@ export function TestFlow(props: TestFlowProps) {
       .unwrap()
       .then((response) => {
         setSearchResponse(response);
+
+        // persist a new memory ID to be optionally injected into the query from the user.
+        const respMemoryId = response?.ext?.memory_id ?? '';
+        const existingMemoryId = finalQuery?.query?.agentic?.memory_id ?? '';
+        if (respMemoryId !== '' && existingMemoryId !== respMemoryId) {
```

Comment:
finally we don't have to check
```suggestion
        if (respMemoryId && existingMemoryId !== respMemoryId) {
```